src/wms/src/main/java/org/geoserver/wms/capabilities/DimensionHelper.java

```diff
@@ -504,19 +508,22 @@ protected String getNumberRepresentation(
             } else {
                 if (values.size() >= 2 && allNumbers(values)) {
                     int count = 2, i = 2;
-                    Double[] zPositions = new Double[count];
+                    Number[] zPositions = new Number[count];
                     // convert all to double
-                    final List<Double> doubleValues =
-                            values.stream()
-                                    .map(x -> (Number) x)
-                                    .map(Number::doubleValue)
-                                    .collect(Collectors.toList());
-                    for (Object val : doubleValues) {
-                        zPositions[count - i--] = (Double) val;
+                    final List<Number> numberValues =
+                            values.stream().map(x -> (Number) x).collect(Collectors.toList());
+                    for (Object val : numberValues) {
+                        zPositions[count - i--] = (Number) val;
                         if (i == 0) break;
                     }
-                    double span = zPositions[count - 1] - zPositions[count - 2];
-                    buff.append(span);
+                    if (isDecimal)
+                        buff.append(
+                                zPositions[count - 1].doubleValue()
```

Comment:
Not sure what's going on here, why are spans being computed? Dimensions either report single values or min/max of every internal, never seen them report the span of an interval... also check the WMS specification about it.

Reply:
Looks like this is a discrete interval representation with a min/max/resolution format.